integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java

```diff
@@ -0,0 +1,119 @@
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import jakarta.inject.Inject;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.awaitility.Awaitility;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class OTelJaegerRemoteTest {
+
+    private static final OkHttpClient client = new OkHttpClient();
+    private static final int QUERY_PORT = 16686;
+    private static final int COLLECTOR_PORT = 14250;
+    private static final int HEALTH_PORT = 14269;
+    private static final String JAEGER_URL = "http://localhost";
+    @Inject
+    OpenTelemetry openTelemetry;
+    private static final DockerClient dockerClient = DockerClientBuilder.getInstance().build();
+
+    static {
+        if(dockerClient.listContainersCmd().exec().stream().noneMatch(container -> container.getNames()[0].equals("/jaeger"))) {
+            ExposedPort queryPort = ExposedPort.tcp(QUERY_PORT);
+            ExposedPort collectorPort = ExposedPort.tcp(COLLECTOR_PORT);
+            ExposedPort hostPort = ExposedPort.tcp(HEALTH_PORT);
+            Ports portBindings = new Ports();
+            portBindings.bind(queryPort, Ports.Binding.bindPort(QUERY_PORT));
+            portBindings.bind(collectorPort, Ports.Binding.bindPort(COLLECTOR_PORT));
+            portBindings.bind(hostPort, Ports.Binding.bindPort(HEALTH_PORT));
+            CreateContainerResponse container = dockerClient.createContainerCmd("ghcr.io/open-telemetry/opentelemetry-java/jaeger:1.32")
+                    .withExposedPorts(queryPort, collectorPort, hostPort)
+                    .withHostConfig(newHostConfig().withPortBindings(portBindings))
+                    .withName("jaeger")
+                    .exec();
+            dockerClient.startContainerCmd(container.getId()).exec();
+        }
+    }
+
+    @AfterAll
+    static void teardown() {
+        dockerClient.listContainersCmd().exec()
+                .forEach(container -> {
+                    dockerClient.stopContainerCmd(container.getId()).exec();
+                    dockerClient.removeContainerCmd(container.getId()).exec();
+                });
+    }
+
+    @Test
+    void testJaegerRemoteIntegration() {
+        createTestSpan(openTelemetry);
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(30))
+                .until(OTelJaegerRemoteTest::assertJaegerHaveTrace);
+    }
+
+    private void createTestSpan(OpenTelemetry openTelemetry) {
+        Span span =
+                openTelemetry.getTracer(getClass().getCanonicalName()).spanBuilder("Test span").startSpan();
+        span.addEvent("some event");
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        span.end();
```

Comment:
Please take a look on how to create manual spans, here: https://opentelemetry.io/docs/instrumentation/java/manual/#create-spans
try/finally block is needed with an OTel Scope.
You don't need the sleep.

Reply:
Thanks for the review @brunobat. I'm working on the other things, meanwhile, I'm kind of stuck on one issue. Basically, I'm trying to test Jaeger autoconfiguration and it needs a docker container to set the exporter endpoint correctly. So I'm creating a test container in the tests but since the configurations are set before the initialization of the container in the test, it is not able to set the exporter properly, resulting in flaky tests. Can you guide what could be done here?

Reply:
Have done changes after referring this.